SonarQube.Common/CommandLine/ArgumentDescriptor.cs

```diff
@@ -14,7 +14,11 @@ namespace SonarQube.Common
     /// </summary>
     public class ArgumentDescriptor
     {
-        public static readonly StringComparer IdComparer = StringComparer.InvariantCultureIgnoreCase;
+        // https://msdn.microsoft.com/en-us/library/ms973919.aspx
+        // "[d]ata that is designed to be culture-agnostic and linguistically irrelevant should... 
+        //  use either StringComparison.Ordinal or StringComparison.OrdinalIgnoreCase..."
+        public static readonly StringComparer IdComparer = StringComparer.Ordinal;
```

Comment:
There are now five(AnalysisSetting key and value, Property key, ArgumentDescriptor Id, AnalysisResult key). It seems reasonable to have a single standard id/key comparer if we're sure want the same rule to apply to all ids. I'll change it.